curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -172,16 +172,18 @@ bool DiskCacheManager::IsCached(const std::string name) {
     return true;
 }
 
+bool DiskCacheManager::IsCacheClean() {
+    return cacheWrite_->IsCacheClean();
+}
+
 int DiskCacheManager::UmountDiskCache() {
     LOG(INFO) << "umount disk cache.";
-    int ret;
     diskInitThread_.join();
-    ret = cacheWrite_->UploadAllCacheWriteFile();
-    if (ret < 0) {
-        LOG(ERROR) << "umount disk cache error.";
-    }
```

Comment:
You can consider dropping the empty task here and wait for the empty task to complete.

The premise of this is `taskPool_.Stop(); ` 
https://github.com/opencurve/curve/blob/9d5a18a547ce689a33668fa936271606b4ecd9d9/curvefs/src/client/s3/disk_cache_manager_impl.cpp#L195
before `diskCacheManager_->UmountDiskCache(); `
https://github.com/opencurve/curve/blob/9d5a18a547ce689a33668fa936271606b4ecd9d9/curvefs/src/client/s3/disk_cache_manager_impl.cpp#L190

Reply:
Right, putting `taskPool_.Stop()` first makes sure `AsyncUploadEnqueue` will be never called, but what's the empty task?